Automattic/jetpack
# VideoPress Package: Refactor useSyncMedia hook

Refactor part of preparation for implementing a native version of the `useSyncMedia` hook. Note, this will be the first of a series of planned PRs for splitting the hooks functionality into smaller hooks, with the goal of making it more straightforward to implement the native functionality. 

See https://github.com/wordpress-mobile/gutenberg-mobile/issues/5611 for more context.

## Proposed changes:

* With this initial PR, the `useSyncMedia` hook has been moved from the `hooks/use-video-data-update/index.ts` file to its own  `hooks/use-sync-media/index.ts` file.
* The `useMediaUpdate` hook remains in place at `hooks/use-video-data-update/index.ts`.
* The READMEs for both hooks have been updated as all imports of `useSyncMedia` that exist under the VideoPress package. 

## Other information:


## Does this pull request change what data or activity we track or use?

No, it does not.

## Jetpack product discussion

p9ugOq-3mV-p2 

## Testing instructions:

> **Note**
> Note, when searching the codebase for `useSyncMedia`, you'll see that there is a hook for the older VideoPress block under the Jetpack plugin's codebase. We should not touch or change the imports related to this.

To test, we should ensure all relevant areas of the codebase have been updated and ensure there are no unwanted side effects when it comes to media syncing:

* Navigate to the block editor.
* Add the VideoPress block and upload a new video.
* Make some changes to settings that should be synced by default e.g. privacy settings or chapters.
* Navigate to the VideoPress editor via _Jetpack_ → _VideoPress_ and ensure changes are synced as expected.
